### PR TITLE
ssr: use `apply_type ~typecheck:true` everywhere (fix #6634)

### LINKED ISSUE #6634: Incorrect generalization
#### Version

8.7.0

#### Operating system

Ubuntu 16.04

#### Description of the problem

Consider the following snippet:

```
From Coq Require Import ssreflect.

Lemma normalizeP (p : tt = tt) : p = p.
Proof.
move: {2} tt p. (* Weird *)
Abort.
```

After executing the first line, the goal becomes the ill-typed term

```
forall (u : unit) (p : @eq unit tt u), @eq (@eq unit tt tt) p p
```

That is, the type of `p` does not match the type expected by `@eq (@eq unit tt tt)`. I expected this to be caught by the generalization in `move`.

### PR DESCRIPTION
**Pull request by gares**
Thanks @arthuraa for reporting

### CODE REVIEW

File: plugins/ssr/ssrelim.ml [hunk 1]
```diff
@@ -28,7 +28,7 @@ module RelDecl = Context.Rel.Declaration
 
 (** The "case" and "elim" tactic *)
 
-let apply_type x xs = Proofview.V82.of_tactic (Tactics.apply_type ~typecheck:false x xs)
+let apply_type x xs = Proofview.V82.of_tactic (Tactics.apply_type ~typecheck:true x xs)
```
Comment (maximedenes) [inline plugins/ssr/ssrelim.ml]: Why redefine this here? Isn't it the same as `Ssrcommon.apply_type`?

**Comment by maximedenes** [comment]
Is it never the case that ssr's code calls `apply_type` on something that is known to be well typed?

File: plugins/ssr/ssrelim.ml (hunk 1 above)
Comment (gares) [reply plugins/ssr/ssrelim.ml]: I guess so, and I'm afraid I'm not the one that defined apply_type in every file ;-)

**Comment by gares** [comment]
> Is it never the case that ssr's code calls apply_type on something that is known to be well typed?

I did not dare check. According to the bench, the slow case is exactly the one for which I want type checking to be enabled, so... let's leave some room for future optimizations

**Comment by gares** [comment]
FTR, this patch is subsumed by #6700 so, unless that one is not merged for 8.8, this one can be ignored.

**Comment by gares** [comment]
It is not really a dependency, but this PR should be merged after #6700 (that will generate conflicts)

**Pull request merged**